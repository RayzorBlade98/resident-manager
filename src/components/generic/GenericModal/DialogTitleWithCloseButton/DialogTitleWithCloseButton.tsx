import CloseIcon from '@mui/icons-material/Close';
import { DialogTitle, IconButton, Theme } from '@mui/material';
import React from 'react';

const styles = {
  closeButton: {
    position: 'absolute',
    right: 8,
    top: 8,
    color: (theme: Theme) => theme.palette.grey[500],
  },
};

interface DialogTitleWithCloseButtonProps {
  onClose: () => void;
}

/**
 * Dialog title component that has a close button
 */
function DialogTitleWithCloseButton(
  props: React.PropsWithChildren<DialogTitleWithCloseButtonProps>,
) {
  return (
    <DialogTitle>
      {props.children}
      <IconButton
        aria-label="close"
        onClick={props.onClose}
        sx={styles.closeButton}
      >
        <CloseIcon />
      </IconButton>
    </DialogTitle>
  );
}

export default DialogTitleWithCloseButton;
