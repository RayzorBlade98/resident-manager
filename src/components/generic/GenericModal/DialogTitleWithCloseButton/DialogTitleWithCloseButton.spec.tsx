import { RenderResult, fireEvent, render } from '@testing-library/react';
import React from 'react';
import DialogTitleWithCloseButton from './DialogTitleWithCloseButton';

describe('DialogTitleWithCloseButton', () => {
  let onCloseMock: jest.Mock;
  let renderResult: RenderResult;

  beforeAll(() => {
    onCloseMock = jest.fn();
    renderResult = render(<DialogTitleWithCloseButton onClose={onCloseMock} />);
  });

  test('should call onClose when clicking the close button', () => {
    // Act
    const button = renderResult.getByRole('button')!;
    fireEvent.click(button);

    // Assert
    expect(onCloseMock).toHaveBeenCalledTimes(1);
  });
});
