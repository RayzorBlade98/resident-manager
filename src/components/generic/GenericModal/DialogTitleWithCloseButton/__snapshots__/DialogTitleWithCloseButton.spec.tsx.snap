// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialogTitleWithCloseButton should match snapshot 1`] = `<div />`;
